### ISSUE TITLE
Custom MessagePack Resolvers

### ISSUE THREAD

**Issue by vpaulino** [aspnet/SignalR#2370]
Hi there

I have been doing some testing with this implementation  https://github.com/neuecc/MessagePack-CSharp  and its all much familiar with the one found on signalR. Is it the same ? 

What is the correct way to register custom formatters or resolvers of formatters in signalR ? I couldn't find any samples of that on the unit tests.

Best Regards
Vítor Paulino

**Comment by anurse** [comment]
You're correct, we are using MessagePack-CSharp. You can configure custom formatter resolvers by providing an "options" delegate to your call to `AddMessagePackProtocol()`:

```csharp
services.AddSignalR()
    .AddMessagePackProtocol(options => options.FormatterResolvers.Add(...));
```

Closing this as the question has been answered. Feel free to comment further if you need clarification on something! We can always re-open the issue if there's something to fix here.
